client/packages/common/src/hooks/useElectronClient/index.ts

```diff
@@ -0,0 +1,87 @@
+import { useState, useEffect } from 'react';
+
+const DISCOVERY_TIMEOUT = 5000;
+
+// Should match server/server/src/discovery.rs (FrontEndHost)
+export type FrontEndHost = {
+  protocol: 'http' | 'https';
+  port: number;
+  ip: string;
+  name: string;
+  clientVersion: string;
+  // This one is set by electron.ts
+  isLocal: boolean;
+};
+
+type ServerDiscovered = (event: object, value: FrontEndHost) => void;
+export type ElectronAPI = {
```

Comment:
not sure if this is the best place for types to be shared, though it's nice to have it closer to the logic